deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/codegen/OpenApiGeneratorCodeGenBase.java

```diff
@@ -209,21 +237,21 @@ private static OpenApiClientGeneratorWrapper createGeneratorWrapper(Path openApi
 
     private String getBasePackage(final Config config, final Path openApiFilePath) {
         return config
-                .getOptionalValue(getBasePackagePropertyName(openApiFilePath), String.class)
+                .getOptionalValue(getSpecConfigName(BASE_PACKAGE, openApiFilePath), String.class)
                 .orElse(String.format("%s.%s", DEFAULT_PACKAGE, getSanitizedFileName(openApiFilePath)));
     }
 
     private Optional<String> getInputBaseDirRelativeToModule(final Path sourceDir, final Config config) {
-        return config.getOptionalValue(INPUT_BASE_DIR, String.class).map(inputBaseDir -> {
+        return config.getOptionalValue(getGlobalConfigName(INPUT_BASE_DIR), String.class).map(inputBaseDir -> {
             int srcIndex = sourceDir.toString().lastIndexOf("src");
             return srcIndex < 0 ? null : sourceDir.toString().substring(0, srcIndex) + inputBaseDir;
         });
     }
 
-    private <T> Optional<T> getValues(final Config config, final Path openApiFilePath, String configName,
+    private <T> Optional<T> getValues(final Config config, final Path openApiFilePath, CodegenConfig.ConfigName configName,
```

Comment:
You're checking if `apiSpecConfigValue` is present and then using `or`.
`Or` does precisely that.

For example:
```java
Optional<String> x = a.or(() -> b);
```
Is identical to:
```java
Optional<String> x;
if (a.isPresent()) {
    x = a;
} else {
    x = b;
}
```

So, in this case you can replace everything with:

```java
    private <T> Optional<T> getValues(final Config config, final Path openApiFilePath, CodegenConfig.ConfigName configName,
            Class<T> propertyType) {
        return config
                .getOptionalValue(CodegenConfig.getSpecConfigName(configName, openApiFilePath), propertyType)
                .or(() -> config.getOptionalValue(CodegenConfig.getGlobalConfigName(configName), propertyType));
    }
```